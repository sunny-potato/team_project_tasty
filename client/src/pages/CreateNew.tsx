import React, { useState } from 'react';
import { RecipeInfo, Ingredient } from '../DataService';
import InputRecipeInfo from '../components/InputRecipeInfo';
import InputIngredients from '../components/InputIngredients';
import { useNavigate } from 'react-router-dom';
import { ValidateData, SaveEditedData } from '../components/SaveInputData';

const initialRecipeInfo: RecipeInfo = {
  id: undefined!,
  name: '',
  meal_type: '',
  new: false,
  popular: false,
  description: '',
};

const initialIngredient: Ingredient = {
  ingredients_id: undefined!,
  ingredient: '',
  amount: null,
  unit_id: 0,
  unit: 'initalUnit',
};

export function CreateNew() {
  const navigate = useNavigate();
  const [recipeInfo, setRecipeInfo] = useState<RecipeInfo>(initialRecipeInfo);
  const [ingredients, setIngredients] = useState<Ingredient[]>([initialIngredient]);
  const [isLoading, setIsLoading] = useState<boolean>(false);

  // console.log(recipeInfo);
  // console.log(ingredients);
  const onSubmit = (event: React.FormEvent) => {
    event.preventDefault();

    if (!isLoading) {
      setIsLoading(true);
      const isValid = ValidateData({ ingredients }); // current condition is always true
      if (isValid) {
        SaveEditedData({ ingredients, recipeInfo });
      }
    } else {
      alert('something wrong');
      //*********************//
      //here will be added f meal_type or description is required
    }
    setIsLoading(false);
  };

  return (
    <div>
      <h1 className="Page-title">Create new recipe</h1>
      <div className="Content-main">
        <form onSubmit={onSubmit}>
          <InputRecipeInfo recipeInfo={recipeInfo} setRecipeInfo={setRecipeInfo} />
          <InputIngredients ingredients={ingredients} setIngredients={setIngredients} />
          <div className="btn-group">
            <button type="submit" style={{ opacity: isLoading ? 0.3 : 1 }}>
              {isLoading ? 'Saving' : 'Save'}
            </button>
            <button type="button" className="cancelButton" onClick={() => navigate('/')}>
              Cancel
            </button>
          </div>
        </form>
      </div>
    </div>
  );
}
