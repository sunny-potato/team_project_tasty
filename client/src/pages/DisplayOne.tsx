import './PageStyling.css';
import React, { useEffect, useState, useRef, Component } from 'react';
import dataService, { Recipe, Ingredient, RecipeInfo } from '../DataService';
import { Link, useParams, useLocation } from 'react-router-dom';

export function DisplayOne() {
  const [recipe, setRecipe] = useState<Recipe>();
  const [ingredients, setIngredients] = useState<Ingredient[]>();

  //To get navigation by page and id
  const params = useParams();
  const id: string = params.id;

  // Get inital value in load, uses recipe id as source
  // Also loads values recipe and ingredients by setState

  useEffect(() => {
    let id: string = params.id;

    dataService.get(id).then((data) => {
      setRecipe(data);
      setIngredients(data.ingredients);
    });
  }, []);

  // Updates ingredients according to value chosen by user

  function changePortions(portions: number) {
    let newAmounts: Ingredient[];
    let defaultAmounts: Ingredient[] = recipe!.ingredients;

    newAmounts = defaultAmounts?.map((i) => ({
      ...i,
      amount: parseFloat(((i.amount / 4) * portions).toPrecision(2)),
    }));
    setIngredients(newAmounts);
  }

  return (
    <div className="Content-main">
      <h1 className="Recipe-title">{recipe?.recipeInfo.name}</h1>
      <h5 className="Recipe-tags">Tags: {recipe?.recipeInfo.meal_type}</h5>
      <h6 className="Recipe-new">{recipe?.recipeInfo.new ? 'This is a new recipe!' : null}</h6>
      <h6 className="Recipe-popular">
        {recipe?.recipeInfo.popular ? 'This item is popular right now!' : null}
      </h6>
      <p className="Recipe-description>">{recipe?.recipeInfo.description}</p>
      <div>
        <div>
          <label>Number of portions:&nbsp;</label>
          <select
            name="portions"
            defaultValue={4}
            onChange={(e) => changePortions(parseInt(e.currentTarget.value))}
          >
            <option value={1}>1</option>
            <option value={2}>2</option>
            <option value={3}>3</option>
            <option value={4}>4</option>
            <option value={5}>5</option>
            <option value={6}>6</option>
            <option value={7}>7</option>
            <option value={8}>8</option>
            <option value={9}>9</option>
            <option value={10}>10</option>
          </select>
        </div>
        <table className="Ingredients-main">
          <tbody>
            <tr>
              <th className="Ingredients-header">Ingredients:</th>
              <th></th>
            </tr>
            {ingredients?.map((content) => (
              <tr className="Ingredients-row" key={content.ingredients_id}>
                <td className="Ingredients-cell">
                  {content.amount}&nbsp;{content.unit}
                </td>

                <td className="Ingredients-name">{content.ingredient}</td>
              </tr>
            ))}
          </tbody>
        </table>
        <Link to="/">
          <button className="Button-navigation">Back</button>
        </Link>
        <Link to="/edit">
          <button className="Button-navigation">Edit recipe</button>
        </Link>
        <button
          className="Button-navigation"
          onClick={() => alert('Here will the like function be, not implemented yet')}
        >
          Like
        </button>
      </div>
    </div>
  );
}
