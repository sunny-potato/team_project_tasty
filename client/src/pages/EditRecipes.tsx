import React from 'react';
import { useParams } from 'react-router-dom';

export function EditRecipes() {
  return (
    <div>
      <h1>Site to edit the recipes</h1>
    </div>
  );
}
