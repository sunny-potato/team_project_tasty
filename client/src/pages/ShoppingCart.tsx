import React from 'react';

export function ShoppingCart() {
  return (
    <div>
      <h1>This is your shopping cart to be....</h1>
    </div>
  );
}
