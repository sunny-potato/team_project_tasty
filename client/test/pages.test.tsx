import * as React from 'react';
import dataService, { Recipe, RecipeInfo, Ingredient } from '../src/DataService';
import { shallow } from 'enzyme';
import { MemoryRouter, NavLink } from 'react-router-dom';
import { Home } from '../src/pages/Home';
import { DisplayOne } from '../src/pages/DisplayOne';
import { EditRecipes } from '../src/pages/EditRecipes';
import Explore from '../src/pages/Explore';
import { ShoppingCart } from '../src/pages/ShoppingCart';
import { CreateNew } from '../src/pages/CreateNew';

import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import { tupleExpression } from '@babel/types';
import { screen, fireEvent } from '@testing-library/react';
import { BsPrinter } from 'react-icons/bs';

let container: any = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

// mock axios api to server
jest.mock('../src/DataService', () => {
  class DataService {
    /* Get a specific recipe with known id */
    get(id: string) {
      return Promise.resolve({
        recipeInfo: {
          id: 1,
          name: 'One-Pot Mac and Cheese',
          meal_type: 'Dinner',
          new: true,
          popular: true,
          description: `Who likes cleaning up after making mac and cheese? 
              Not this girl. This one-pot mac and cheese is a family favorite, 
              and my 3-year-old is thrilled to see it coming to the dinner table. 
              We love to add sliced smoked sausage to this creamy mac recipe!`,
        },
        ingredients: [
          {
            ingredients_id: 1,
            ingredient: 'whole milk',
            amount: 1.5,
            unit_id: 2,
            unit: 'cup',
          },
          {
            ingredients_id: 2,
            ingredient: 'water',
            amount: 3,
            unit_id: 2,
            unit: 'cup',
          },
          {
            ingredients_id: 3,
            ingredient: 'elbow macaroni (16 ounces)',
            amount: 1,
            unit_id: 3,
            unit: 'package',
          },
          {
            ingredients_id: 4,
            ingredient: 'velveeta, cubed',
            amount: 4,
            unit_id: 4,
            unit: 'ounce',
          },
          {
            ingredients_id: 5,
            ingredient: 'shredded sharp cheddar cheese',
            amount: 2,
            unit_id: 2,
            unit: 'cup',
          },
          {
            ingredients_id: 6,
            ingredient: 'salt',
            amount: 0.5,
            unit_id: 5,
            unit: 'teaspoon',
          },
          {
            ingredients_id: 7,
            ingredient: 'coarsely ground pepper',
            amount: 0.5,
            unit_id: 5,
            unit: 'teaspoon',
          },
        ],
      });
    }

    // Get an array of all recipes
    getAll() {
      return Promise.resolve([
        {
          id: 1,
          name: 'One-Pot Mac and Cheese',
          meal_type: 'Dinner',
          new: true,
          popular: true,
          description: `Who likes cleaning up after making mac and cheese? 
          Not this girl. This one-pot mac and cheese is a family favorite, 
          and my 3-year-old is thrilled to see it coming to the dinner table. 
          We love to add sliced smoked sausage to this creamy mac recipe!`,
        },
        {
          id: 2,
          name: 'Easy Marinated Grilled Flank Steak',
          meal_type: 'Dinner',
          new: false,
          popular: false,
          description: `Friends shared this three-ingredient marinade years ago, 
          and itâ€™s been a favorite since. Serve the steak with salad and 
          grilled potatoes for a quick meal.`,
        },
      ]);
    }

    /* Post new recepie */
    create() {
      return Promise.resolve(3); // Same as: return new Promise((resolve) => resolve(3));
    }

    edit() {
      return Promise.resolve();
    }

    delete() {
      return Promise.resolve();
    }
  }
  return new DataService();
});

// mock localStorage
const localStorageMock = (() => {
  let cart: Ingredient[] = [
    { ingredients_id: 1, ingredient: 'whole milk', amount: 1.5, unit_id: 2, unit: 'cup' },
    { ingredients_id: 2, ingredient: 'water', amount: 3, unit_id: 2, unit: 'cup' },
  ];

  return {
    getItem(key: string) {
      console.log(JSON.stringify(cart));
      return JSON.stringify(cart);
    },

    setItem(key: string, value: string) {
      cart = JSON.parse(value);
    },

    clear() {
      cart = [];
    },

    removeItem(key: string) {},

    getAll() {
      return cart;
    },
  };
})();
Object.defineProperty(window, 'localStorage', { value: localStorageMock });

describe('Home page tests', () => {
  test('Home page includes correct static elements', (done) => {
    const wrapper = shallow(<Home />);
    setTimeout(() => {
      expect(
        wrapper.containsAllMatchingElements([
          <h1>Welcome to the Tasty</h1>,
          <p>
            Here you can explore your own recipes by adding new content, editing existing ones and
            deleting the ones you don't care about anymore.
          </p>,
          <p>
            This is also the place to discover more and get inspiration from, we have selected a few
            to show you from [External API] below.
          </p>,
          <div className="Content-second"></div>,
        ])
      ).toEqual(true);
      done();
    });
  });
  test('Home page draws correctly (snapshot) including list of buttons', async () => {
    await act(async () => {
      await render(
        <MemoryRouter>
          <Home />
        </MemoryRouter>,
        container
      );
    });
    expect(container).toMatchSnapshot();
  });
});

describe('DisplayOne page tests', () => {
  test('DisplayOne page draws correctly (snapshot) including data from test recipe 1', async () => {
    await act(async () => {
      await render(
        <MemoryRouter>
          <DisplayOne />
        </MemoryRouter>,
        container
      );
    });
    expect(container).toMatchSnapshot();
  });

  test('Changing portions', async () => {
    await act(async () => {
      await render(
        <MemoryRouter>
          <DisplayOne />
        </MemoryRouter>,
        container
      );
    });

    // get hold of the current amount on ingredient 1
    let amount1 = container.querySelector('div table td')?.innerHTML.substring(0, 3);
    expect(amount1).toBe('1.5');

    // get hold of selector
    const selector = container.querySelector('div select');

    // change selector to 6 portions (firing event using react test library)
    fireEvent.change(selector, { target: { value: 6 } });

    // update and check new amount on ingredient 1
    amount1 = container.querySelector('div table td')?.innerHTML.substring(0, 3);
    expect(amount1).toBe('2.3');
  });

  test('Like recipe', async () => {
    await act(async () => {
      await render(
        <MemoryRouter>
          <DisplayOne />
        </MemoryRouter>,
        container
      );
    });

    // confirm that this is a popular recipe
    let popular = container.querySelector('div h6.Recipe-popular');
    expect(popular.innerHTML).toBe('This item is popular');

    // click on popular icon (firing event using react test library)
    fireEvent.click(container.querySelector("div button[title='Like the recipe']"));

    // confirm that recipe is no longer popular
    popular = container.querySelector('div h6.Recipe-popular');
    expect(popular.innerHTML).toBe('');
  });
});

describe('EditRecipes page tests', () => {
  test('EditRecipes page draws correctly', (done) => {
    const wrapper = shallow(<EditRecipes />);
    setTimeout(() => {
      expect(wrapper).toMatchSnapshot();
      done();
    });
  });
});

describe('Explore page tests', () => {
  test('Explore page draws correctly', (done) => {
    const wrapper = shallow(<Explore />);
    setTimeout(() => {
      expect(wrapper).toMatchSnapshot();
      done();
    });
  });
});

describe('ShoppingCart page tests', () => {
  test('ShoppingCart page draws correctly', async () => {
    await act(async () => {
      await render(
        <MemoryRouter>
          <ShoppingCart />
        </MemoryRouter>,
        container
      );
    });
    expect(container).toMatchSnapshot();
  });
  test('Increase and decrease amount', async () => {
    await act(async () => {
      await render(
        <MemoryRouter>
          <ShoppingCart />
        </MemoryRouter>,
        container
      );
    });
    //get and check current amount on first ingredient
    let amount1 = container.querySelector('td.Ingredients-cell')?.innerHTML;
    expect(amount1).toBe('1.5');

    // increase amount and then check value again
    fireEvent.click(container.querySelector('.Icon-increase'));
    amount1 = container.querySelector('td.Ingredients-cell')?.innerHTML;
    expect(amount1).toBe('1.6');

    // decrease amount by 0.2 and check value again
    fireEvent.click(container.querySelector('.Icon-decrease'));
    fireEvent.click(container.querySelector('.Icon-decrease'));
    amount1 = container.querySelector('td.Ingredients-cell')?.innerHTML;
    expect(amount1).toBe('1.4');
  });
  test('Delete ingredient', async () => {
    await act(async () => {
      await render(
        <MemoryRouter>
          <ShoppingCart />
        </MemoryRouter>,
        container
      );
    });
    // get and check name of first ingredient
    let ingredient1 = container.querySelector('.Ingredients-name')?.innerHTML;
    expect(ingredient1).toBe('whole milk');

    // delete the first ingredient and check that ingredient no longer is 'whole milk'
    fireEvent.click(container.querySelector('.Icon-remove'));
    ingredient1 = container.querySelector('.Ingredients-name')?.innerHTML;
    expect(ingredient1).toBe('water');
  });
});

describe('CreateNew page tests', () => {
  test('CreateNew page draws correctly', (done) => {
    const wrapper = shallow(<CreateNew />);
    setTimeout(() => {
      expect(wrapper).toMatchSnapshot();
      done();
    });
  });
});
