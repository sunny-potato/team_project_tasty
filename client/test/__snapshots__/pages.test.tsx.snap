// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateNew page tests CreateNew page draws correctly 1`] = `
<div>
  <div
    className="Content-main"
  >
    <h1
      className="Content-title"
    >
      Create new recipe
    </h1>
    <p>
      Instructions Lorem ipsum dolor sit amet consectetur, adipisicing elit. Pariatur placeat quasi dolor illum? Assumenda magni architecto velit aspernatur consequuntur cupiditate fugit vero tenetur rem perspiciatis exercitationem harum, ratione quas corrupti.
    </p>
    <div
      className="User-input"
    >
      <input
        placeholder="Enter recipe title"
        type="text"
        value=""
      />
    </div>
    <div
      className="User-input"
    >
      <textarea
        cols={50}
        placeholder="Enter description of recipe"
        rows={12}
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`DisplayOne page tests DisplayOne page draws correctly (snapshot) including data from test recipe 1 1`] = `
<div>
  <div
    class="Content-main"
  >
    <h1
      class="Recipe-title"
    >
      One-Pot Mac and Cheese
    </h1>
    <h5
      class="Recipe-tags"
    >
      Tags: 
      Dinner
    </h5>
    <h6
      class="Recipe-new"
    >
      This is a new recipe!
    </h6>
    <h6
      class="Recipe-popular"
    >
      This item is popular
    </h6>
    <svg
      class="Icon-print"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Print the shopping list
      </title>
      <path
        d="M2.5 8a.5.5 0 1 0 0-1 .5.5 0 0 0 0 1z"
      />
      <path
        d="M5 1a2 2 0 0 0-2 2v2H2a2 2 0 0 0-2 2v3a2 2 0 0 0 2 2h1v1a2 2 0 0 0 2 2h6a2 2 0 0 0 2-2v-1h1a2 2 0 0 0 2-2V7a2 2 0 0 0-2-2h-1V3a2 2 0 0 0-2-2H5zM4 3a1 1 0 0 1 1-1h6a1 1 0 0 1 1 1v2H4V3zm1 5a2 2 0 0 0-2 2v1H2a1 1 0 0 1-1-1V7a1 1 0 0 1 1-1h12a1 1 0 0 1 1 1v3a1 1 0 0 1-1 1h-1v-1a2 2 0 0 0-2-2H5zm7 2v3a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1v-3a1 1 0 0 1 1-1h6a1 1 0 0 1 1 1z"
      />
    </svg>
    <p
      class="Recipe-description>"
    >
      Who likes cleaning up after making mac and cheese? 
              Not this girl. This one-pot mac and cheese is a family favorite, 
              and my 3-year-old is thrilled to see it coming to the dinner table. 
              We love to add sliced smoked sausage to this creamy mac recipe!
    </p>
    <div>
      <div>
        <label>
          Number of portions: 
        </label>
        <select
          name="portions"
          title="Number of portions"
        >
          <option
            value="1"
          >
            1
          </option>
          <option
            value="2"
          >
            2
          </option>
          <option
            value="3"
          >
            3
          </option>
          <option
            selected=""
            value="4"
          >
            4
          </option>
          <option
            value="5"
          >
            5
          </option>
          <option
            value="6"
          >
            6
          </option>
          <option
            value="7"
          >
            7
          </option>
          <option
            value="8"
          >
            8
          </option>
          <option
            value="9"
          >
            9
          </option>
          <option
            value="10"
          >
            10
          </option>
        </select>
      </div>
      <table
        class="Ingredients-main"
      >
        <tbody>
          <tr>
            <th
              class="Ingredients-header"
            >
              Ingredients:
            </th>
            <th />
          </tr>
          <tr
            class="Ingredients-row"
          >
            <td
              class="Ingredients-cell"
            >
              1.5
               
              cup
            </td>
            <td
              class="Ingredients-name"
            >
              whole milk
            </td>
          </tr>
          <tr
            class="Ingredients-row"
          >
            <td
              class="Ingredients-cell"
            >
              3
               
              cup
            </td>
            <td
              class="Ingredients-name"
            >
              water
            </td>
          </tr>
          <tr
            class="Ingredients-row"
          >
            <td
              class="Ingredients-cell"
            >
              1
               
              package
            </td>
            <td
              class="Ingredients-name"
            >
              elbow macaroni (16 ounces)
            </td>
          </tr>
          <tr
            class="Ingredients-row"
          >
            <td
              class="Ingredients-cell"
            >
              4
               
              ounce
            </td>
            <td
              class="Ingredients-name"
            >
              velveeta, cubed
            </td>
          </tr>
          <tr
            class="Ingredients-row"
          >
            <td
              class="Ingredients-cell"
            >
              2
               
              cup
            </td>
            <td
              class="Ingredients-name"
            >
              shredded sharp cheddar cheese
            </td>
          </tr>
          <tr
            class="Ingredients-row"
          >
            <td
              class="Ingredients-cell"
            >
              0.5
               
              teaspoon
            </td>
            <td
              class="Ingredients-name"
            >
              salt
            </td>
          </tr>
          <tr
            class="Ingredients-row"
          >
            <td
              class="Ingredients-cell"
            >
              0.5
               
              teaspoon
            </td>
            <td
              class="Ingredients-name"
            >
              coarsely ground pepper
            </td>
          </tr>
        </tbody>
      </table>
      <button
        class="Button-navigation"
        title="Like the recipe"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6.956 1.745C7.021.81 7.908.087 8.864.325l.261.066c.463.116.874.456 1.012.965.22.816.533 2.511.062 4.51a9.84 9.84 0 0 1 .443-.051c.713-.065 1.669-.072 2.516.21.518.173.994.681 1.2 1.273.184.532.16 1.162-.234 1.733.058.119.103.242.138.363.077.27.113.567.113.856 0 .289-.036.586-.113.856-.039.135-.09.273-.16.404.169.387.107.819-.003 1.148a3.163 3.163 0 0 1-.488.901c.054.152.076.312.076.465 0 .305-.089.625-.253.912C13.1 15.522 12.437 16 11.5 16H8c-.605 0-1.07-.081-1.466-.218a4.82 4.82 0 0 1-.97-.484l-.048-.03c-.504-.307-.999-.609-2.068-.722C2.682 14.464 2 13.846 2 13V9c0-.85.685-1.432 1.357-1.615.849-.232 1.574-.787 2.132-1.41.56-.627.914-1.28 1.039-1.639.199-.575.356-1.539.428-2.59z"
          />
        </svg>
      </button>
      <a
        href="/edit/undefined"
      >
        <button
          class="Button-navigation"
          title="Edit recipe"
        >
          Edit recipe
        </button>
      </a>
      <a
        href="/cart"
      >
        <button
          class="Button-navigation"
          title="Add recipe to shopping list"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M4 6.414L.757 3.172l1.415-1.415L5.414 5h15.242a1 1 0 0 1 .958 1.287l-2.4 8a1 1 0 0 1-.958.713H6v2h11v2H5a1 1 0 0 1-1-1V6.414zM6 7v6h11.512l1.8-6H6zm-.5 16a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm12 0a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3z"
              />
            </g>
          </svg>
           
        </button>
      </a>
    </div>
    <div>
      <a
        href="/"
      >
        <button
          class="Button-navigation"
          title="Back"
        >
          Back
        </button>
      </a>
    </div>
  </div>
</div>
`;

exports[`EditRecipes page tests EditRecipes page draws correctly 1`] = `
<div>
  <h1>
    Site to edit the recipes
  </h1>
</div>
`;

exports[`Explore page tests Explore page draws correctly 1`] = `
<div
  className="Content-main"
>
  <h1>
    A place to discover new recipes, API or other
  </h1>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Repudiandae, perferendis illo nam consectetur adipisci quas quaerat omnis dolore repellendus deserunt, at placeat, laudantium quod? Facilis, nam soluta voluptates modi odio, nostrum deleniti cupiditate molestias debitis iste, error laboriosam? Officia fugit quos officiis voluptatem iure laboriosam perferendis libero sit sapiente impedit.
  </p>
</div>
`;

exports[`Home page tests Home page draws correctly (snapshot) including list of buttons 1`] = `
<div>
  <div
    class="Content-main"
  >
    <h1>
      Welcome to the Tasty
    </h1>
    <p>
      Here you can explore your own recipes by adding new content, editing existing ones and deleting the ones you don't care about anymore.
    </p>
    <p>
      This is also the place to discover more and get inspiration from, we have selected a few to show you from [External API] below.
    </p>
    <div
      class="Content-second"
    >
      <div>
         
        <a
          href="/recipe/1"
        >
          <button
            class="Button-recipe-navigation"
          >
            One-Pot Mac and Cheese
          </button>
        </a>
        <a
          href="/recipe/2"
        >
          <button
            class="Button-recipe-navigation"
          >
            Easy Marinated Grilled Flank Steak
          </button>
        </a>
      </div>
      <div
        class="Content-second"
      />
    </div>
  </div>
</div>
`;

exports[`ShoppingCart page tests ShoppingCart page draws correctly 1`] = `
<div>
  <div
    class="Content-main"
  >
    <h1>
      Shopping list
    </h1>
    <svg
      class="Icon-print"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Print the shopping list
      </title>
      <path
        d="M2.5 8a.5.5 0 1 0 0-1 .5.5 0 0 0 0 1z"
      />
      <path
        d="M5 1a2 2 0 0 0-2 2v2H2a2 2 0 0 0-2 2v3a2 2 0 0 0 2 2h1v1a2 2 0 0 0 2 2h6a2 2 0 0 0 2-2v-1h1a2 2 0 0 0 2-2V7a2 2 0 0 0-2-2h-1V3a2 2 0 0 0-2-2H5zM4 3a1 1 0 0 1 1-1h6a1 1 0 0 1 1 1v2H4V3zm1 5a2 2 0 0 0-2 2v1H2a1 1 0 0 1-1-1V7a1 1 0 0 1 1-1h12a1 1 0 0 1 1 1v3a1 1 0 0 1-1 1h-1v-1a2 2 0 0 0-2-2H5zm7 2v3a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1v-3a1 1 0 0 1 1-1h6a1 1 0 0 1 1 1z"
      />
    </svg>
    <table
      class="Ingredients-main"
    >
      <tbody>
        <tr>
          <td>
            <svg
              class="Icon-increase"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Increase amount
              </title>
              <path
                d="M363 277h-86v86h-42v-86h-86v-42h86v-86h42v86h86v42z"
              />
              <path
                d="M256 90c44.3 0 86 17.3 117.4 48.6C404.7 170 422 211.7 422 256s-17.3 86-48.6 117.4C342 404.7 300.3 422 256 422c-44.3 0-86-17.3-117.4-48.6C107.3 342 90 300.3 90 256c0-44.3 17.3-86 48.6-117.4C170 107.3 211.7 90 256 90m0-42C141.1 48 48 141.1 48 256s93.1 208 208 208 208-93.1 208-208S370.9 48 256 48z"
              />
            </svg>
          </td>
          <td
            class="Ingredients-cell"
          >
            1.5
          </td>
          <td>
            <svg
              class="Icon-decrease"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Decrease amount
              </title>
              <path
                d="M256 90c44.3 0 86 17.3 117.4 48.6C404.7 170 422 211.7 422 256s-17.3 86-48.6 117.4C342 404.7 300.3 422 256 422s-86-17.3-117.4-48.6C107.3 342 90 300.3 90 256s17.3-86 48.6-117.4C170 107.3 211.7 90 256 90m0-42C141.1 48 48 141.1 48 256s93.1 208 208 208 208-93.1 208-208S370.9 48 256 48z"
              />
              <path
                d="M363 277H149v-42h214v42z"
              />
            </svg>
          </td>
          <td
            class="Shopping-unit"
          >
            cup
          </td>
          <td
            class="Ingredients-name"
          >
            whole milk
          </td>
          <td>
            <svg
              class="Icon-remove"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                dataName="Square Remove"
                id="Square_Remove"
              >
                <g>
                  <path
                    d="M18.437,20.937H5.563a2.5,2.5,0,0,1-2.5-2.5V5.563a2.5,2.5,0,0,1,2.5-2.5H18.437a2.5,2.5,0,0,1,2.5,2.5V18.437A2.5,2.5,0,0,1,18.437,20.937ZM5.563,4.063a1.5,1.5,0,0,0-1.5,1.5V18.437a1.5,1.5,0,0,0,1.5,1.5H18.437a1.5,1.5,0,0,0,1.5-1.5V5.563a1.5,1.5,0,0,0-1.5-1.5Z"
                  />
                  <path
                    d="M13.767,14.477a.5.5,0,0,0,.71-.71L12.707,12l1.77-1.77a.5.5,0,0,0-.71-.7L12,11.3l-1.77-1.77a.5.5,0,0,0-.7.7c.59.59,1.17,1.18,1.77,1.77l-1.77,1.77c-.46.45.25,1.16.7.71L12,12.707Z"
                  />
                </g>
              </g>
            </svg>
          </td>
        </tr>
        <tr>
          <td>
            <svg
              class="Icon-increase"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Increase amount
              </title>
              <path
                d="M363 277h-86v86h-42v-86h-86v-42h86v-86h42v86h86v42z"
              />
              <path
                d="M256 90c44.3 0 86 17.3 117.4 48.6C404.7 170 422 211.7 422 256s-17.3 86-48.6 117.4C342 404.7 300.3 422 256 422c-44.3 0-86-17.3-117.4-48.6C107.3 342 90 300.3 90 256c0-44.3 17.3-86 48.6-117.4C170 107.3 211.7 90 256 90m0-42C141.1 48 48 141.1 48 256s93.1 208 208 208 208-93.1 208-208S370.9 48 256 48z"
              />
            </svg>
          </td>
          <td
            class="Ingredients-cell"
          >
            3
          </td>
          <td>
            <svg
              class="Icon-decrease"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Decrease amount
              </title>
              <path
                d="M256 90c44.3 0 86 17.3 117.4 48.6C404.7 170 422 211.7 422 256s-17.3 86-48.6 117.4C342 404.7 300.3 422 256 422s-86-17.3-117.4-48.6C107.3 342 90 300.3 90 256s17.3-86 48.6-117.4C170 107.3 211.7 90 256 90m0-42C141.1 48 48 141.1 48 256s93.1 208 208 208 208-93.1 208-208S370.9 48 256 48z"
              />
              <path
                d="M363 277H149v-42h214v42z"
              />
            </svg>
          </td>
          <td
            class="Shopping-unit"
          >
            cup
          </td>
          <td
            class="Ingredients-name"
          >
            water
          </td>
          <td>
            <svg
              class="Icon-remove"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                dataName="Square Remove"
                id="Square_Remove"
              >
                <g>
                  <path
                    d="M18.437,20.937H5.563a2.5,2.5,0,0,1-2.5-2.5V5.563a2.5,2.5,0,0,1,2.5-2.5H18.437a2.5,2.5,0,0,1,2.5,2.5V18.437A2.5,2.5,0,0,1,18.437,20.937ZM5.563,4.063a1.5,1.5,0,0,0-1.5,1.5V18.437a1.5,1.5,0,0,0,1.5,1.5H18.437a1.5,1.5,0,0,0,1.5-1.5V5.563a1.5,1.5,0,0,0-1.5-1.5Z"
                  />
                  <path
                    d="M13.767,14.477a.5.5,0,0,0,.71-.71L12.707,12l1.77-1.77a.5.5,0,0,0-.71-.7L12,11.3l-1.77-1.77a.5.5,0,0,0-.7.7c.59.59,1.17,1.18,1.77,1.77l-1.77,1.77c-.46.45.25,1.16.7.71L12,12.707Z"
                  />
                </g>
              </g>
            </svg>
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="Extra-information"
    >
      Total number of items: 2
    </div>
    <div>
      <a
        href="/"
      >
        <button
          class="Button-navigation"
        >
          Back
        </button>
      </a>
      <button
        class="Button-navigation"
        title="Clear shopping list"
      >
        Clear list
      </button>
    </div>
  </div>
</div>
`;
